contracts/RIFScheduler.sol

```diff
@@ -279,16 +280,31 @@ contract RIFScheduler is IERC677TransferReceiver, ReentrancyGuard, Pausable {
   // CANCELLING //
   ////////////////
 
+  function refund(Execution memory execution) private {
+    remainingExecutions[execution.requestor][execution.plan]++;
+    (bool paymentSuccess, bytes memory paymentResult) = payable(execution.requestor).call{ value: execution.value }('');
+    require(paymentSuccess, string(paymentResult));
+  }
+
+  // This method is meant to be executed by the requestor, to cancel scheduled executions.
+  // If it's called by anyone else, the caller will spend gas and get no benefit.
   function cancelScheduling(bytes32 id) external {
-    require(executions[id].state == ExecutionState.Scheduled, 'Transaction not scheduled'); // Checking state directly to consider Scheduled and Overdue
+    require(getState(id) == ExecutionState.Scheduled, 'Not scheduled');
     require(msg.sender == executions[id].requestor, 'Not authorized');
-
     Execution storage execution = executions[id];
     execution.state = ExecutionState.Cancelled;
-    remainingExecutions[execution.requestor][execution.plan] += 1;
     emit ExecutionCancelled(id);
-    (bool paymentSuccess, bytes memory paymentResult) = payable(execution.requestor).call{ value: execution.value }('');
-    require(paymentSuccess, string(paymentResult));
+    refund(execution);
+  }
+
+  // This method is meant to be executed by the requestor, to get a refund on overdue executions.
+  // If it's called by anyone else, the caller will spend gas and get no benefit.
+  function requestExecutionRefund(bytes32 id) external {
+    require(getState(id) == ExecutionState.Overdue, 'Not overdue');
```

Comment:
Also check if who calls this method is the requester of the scheduled transaction. I as the requester may want to leave the funds there to use in other operations, and if somebody is requesting a refund (even that its sent to me)  can be annoying.

Reply:
Added the required to check this